service/implementation/email_sender/user_subscription_manager.py

```diff
@@ -0,0 +1,52 @@
+import os
+from flask import current_app
+from database.models import TempSubscribersData
+from pathlib import Path
+from dotenv import load_dotenv
+from sqlalchemy import event
+from jinja2 import Environment, FileSystemLoader
+from flask_mail import Message
+from dto.common_response import CommonResponse
+
+class UserSubscriptionManager:
+    def __init__(self, user_subscription_dal):
+        self.__user_subscription_dal = user_subscription_dal
+        current_dir = Path(__file__).resolve().parent if "__file__" in locals() else Path.cwd()
+        envars = current_dir / ".env"
+        load_dotenv(envars)
+
+        event.listen(TempSubscribersData, "after_insert", self.__on_subscribers_inserted)
+
+    def try_add_subscribers_to_temp_table(self, team_index, news_name):
+        self.__user_subscription_dal.try_add_subscribers_data(team_index, news_name)
+
+    def __on_subscribers_inserted(self, mapper, connection, target):
+        self.__try_send_email_to_users()
+
+    def __try_send_email_to_users(self):
+        subscribed_users = self.__user_subscription_dal.pop_subscribed_users_data()
+
+        for user in subscribed_users:
```

Comment:
We should use some sort of Parallel Execution here.

The best approach would be to just use bulk_send emails but I am not sure whether Gmail supports that in the way we need (1. personalized emails 2. user A shouldn't know about user B getting the same email). So, yeah, at least we should make this loop run faster with threads